src/cockpit/superuser.py

```diff
@@ -40,7 +40,10 @@ def __init__(self, router: Router, config: Dict[str, object], responder: ferny.I
     async def do_connect_transport(self) -> asyncio.Transport:
         agent = ferny.InteractionAgent(self.responder)
         transport = await self.spawn(self.args, self.env, stderr=agent, start_new_session=True)
-        await agent.communicate()
+        try:
+            await agent.communicate()
+        except ferny.InteractionError as exc:
+            raise PeerError('authentication-failed', message=str(exc))
```

Comment:
That reminds me about that integration test of entering the sudo password wrong three times. Do you want to write one, or want me to?

Reply:
Happy for you to do it, if you want :)

Reply:
Okay, on my todo list now.